.cargo/config

```diff
@@ -5,7 +5,25 @@
 #
 # See: https://github.com/rust-lang/rust/issues/56068
 # See: https://reviews.llvm.org/D74169#1990180
-rustflags = ["-C", "link-arg=-Wl,--compress-debug-sections=zlib-gabi", "-Csymbol-mangling-version=v0", "--cfg=tokio_unstable"]
+#
+# As of Jan 2024 all of the x86-64 hardware we run on supports the V3 micro architecture.
+#
+# See: https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels
+rustflags = [
+    "-C",
+    "link-arg=-Wl,--compress-debug-sections=zlib-gabi",
+    "-Csymbol-mangling-version=v0",
+    "-Ctarget-cpu=x86-64-v3",
+    "--cfg=tokio_unstable",
+]
+
+# As of Jan 2024 all of the Linux based aarch64 hardware we run on supports the Neoverse N1 micro
+# architecture.
+#
+# See: https://en.wikipedia.org/wiki/AWS_Graviton#Graviton2
+# See: https://github.com/aws/aws-graviton-getting-started/blob/2e6293f27f19df560c12c6d526b78c3384513f99/rust.md
+[target."aarch64-unknown-linux-gnu"]
+rustflags = ["-Ctarget-cpu=neoverse-n1", "--cfg=tokio_unstable"]
```

Comment:
This should have all the same rustflags as x86-64 (modulo the different target CPU, of course). The intent was to enable compressed debuginfo for all folks running `cargo build` on Linux, but Cargo requires that we specify a whole target triple here, and we only bothered to do this for Linux/x86_64 previously. But no reason not to eanble compressed debuginfo for Linux/aarch64 too!

Separately, I'm not sure it makes sense to be specifying a specific target CPU here. The configuration here is only used when running `cargo build` locally, in which case I think `target-cpu=generic` (the current default, I imagine) or `target-cpu=native` makes the most sense. 

So: what do you think about specifying `-Ctarget-cpu=native` here?

Reply:
An argument for keeping the configuration here the same as for production builds is that it maybe removes one cause of performance deviation and so may increase the likelihood that we can reproduce performance issues seen in production locally, or find performance issues during local testing rather than when changes have already been deployed to prod.

That said, we probably don't want to encourage benchmarking in local environments too much, given the differences in, e.g. persist setup. So this might actually be an argument for `target-cpu=native` instead.

Reply:
> This should have all the same rustflags as x86-64

Ahhh good catch, updated!

> Separately, I'm not sure it makes sense to be specifying a specific target CPU here. The configuration here is only used when running cargo build locally, in which case I think target-cpu=generic (the current default, I imagine) or target-cpu=native makes the most sense.

When I first read this comment I totally agreed, leaning more towards `target-cpu=generic`. But once a Miri test failure showed up for an unavailable instruction I now think we should keep these settings in-sync with what we build the final production binary with. Enabling CPU features _shouldn't_ ever cause a change in behavior, but on the off chance that it does, it seems beneficial that local builds will mirror production builds, what do you think?

Reply:
I agree with this take. There doesn't seem to be a good reason to make things more complex by having one more way in which production differs from local.

Reply:
> There doesn't seem to be a good reason to make things more complex by having one more way in which production differs from local.

Well, except for the fact that it will break folks building locally if their CPU is older than what we run in production. I'm down to try it, but we should be ready to adjust if someone complains about `cargo run` locally no longer working for them.

Reply:
Yep, we should back off if it turns out somebody is using an old enough machine. x86-64 v3 is from 2015, so I kinda doubt it, but not implausible.